src/PolyhedralGeometry/PolyhedralComplex/constructors.jl

```
@@ -0,0 +1,87 @@
+polyhedra indicated by an incidence matrix, where the columns represent the
+points and the rows represent the polyhedra.
+
+# Examples
+```jldoctest
+julia> IM = IncidenceMatrix([[1,2,3],[1,3,4]])
+2×4 IncidenceMatrix
+[1, 2, 3]
+[1, 3, 4]
+
+
+julia> VR = [0 0; 1 0; 1 1; 0 1]
+4×2 Matrix{Int64}:
+ 0  0
+ 1  0
+ 1  1
+ 0  1
+
+julia> PC = PolyhedralComplex(IM, VR)
+A polyhedral complex in ambient dimension 2
+```
+"""
+function PolyhedralComplex(Incidence::IncidenceMatrix, VR::Union{SubObjectIterator{<:Union{PointVector,PointVector}}, Oscar.MatElem, AbstractMatrix}, far_vertices::Union{Vector{Int}, Nothing} = nothing, L::Union{SubObjectIterator{<:RayVector}, Oscar.MatElem, AbstractMatrix, Nothing} = nothing)
+    VRM = matrix_for_polymake(VR)
+    LM = isnothing(L) || isempty(L) ? Polymake.Matrix{Polymake.Rational}(undef, 0, size(VRM, 2)) : matrix_for_polymake(L)
+
+    # Rays and Points are homogenized and combined and
+    points = homogenize(VRM, 1)
+    # If some vertices are far vertices, give them a leading 0
+    if !isnothing(far_vertices)
+        points[far_vertices,1] .= 0
+    end
+
+    # Lineality is homogenized
+    lineality = homogenize(LM, 0)
+
+    PolyhedralComplex(Polymake.fan.PolyhedralComplex{Polymake.Rational}(
+        POINTS = points,
+        INPUT_LINEALITY = lineality,
+        INPUT_CONES = Incidence,
+    ))
+end
+
+# This makes sure that PolyhedralComplex(maximal_polyhedra(PC)) returns an Oscar PolyhedralComplex,
+PolyhedralComplex(iter::SubObjectIterator{Polyhedron}) = PolyhedralComplex(iter.Obj)
+
+###############################################################################
+###############################################################################
+### Display
+###############################################################################
+###############################################################################
+function Base.show(io::IO, PC::PolyhedralComplex)
+    ad = ambient_dim(PC)
+    if ad == -1.0
```

I don't understand this line of code: (1) it should not be possible to create a polyhedral complex without an ambient dimension. (2) Why is the right hand side `-1.0` (and not an integer)?

---

Could we not make a polyhedral complex just from a Hasse diagram? @YueRen says he needs polyhedral complexes that aren't embedded. We also had polyhedra without embedding. This right hand side stems from the fact that polymake would not error, but instead return `-1.0` if the ambient dimension could not be computed. The code is the same for the polyhedron, maybe this is fixed now, @benlorenz ?

---

I fixed the code in polymake for Cone, Polytope and PolyhedralFan but somehow missed PolyhedralComplex. I will add the missing input property for the upcoming release, after that change the ambient dim method should fail instead of returning a plain perl number. The code tries to compute `FAN_AMBIENT_DIM - 1` which should be an int but in this case it is `undef - 1` which perl evaluates to a float (perl number type) -1. (This is then translated to a julia Float)